sycl/source/CMakeLists.txt

```diff
@@ -21,6 +21,17 @@ function(add_sycl_rt_library LIB_NAME LIB_OBJ_NAME)
   cmake_parse_arguments(ARG "" "XPTI_LIB;IMPLIB_NAME" "COMPILE_OPTIONS;SOURCES" ${ARGN})
 
   add_library(${LIB_OBJ_NAME} OBJECT ${ARG_SOURCES})
+
+  # Common compilation step setup
+
+  target_compile_definitions(
+    ${LIB_OBJ_NAME}
+    PRIVATE
+      __SYCL_INTERNAL_API
+      SYCL2020_DISABLE_DEPRECATION_WARNINGS
+      $<$<BOOL:${MSVC}>:__SYCL_BUILD_SYCL_DLL>
```

Comment:
Note for reviewers, even though technically it is conditional, I still consider this as common setup. I only consider things like `SYCL_ENABLE_STACK_PRINTING` to be non-common, i.e. thing which can be tweaked by options and not OS-specific things